health/health.go

```diff
@@ -3,26 +3,44 @@
 package health
 
 import (
-	"time"
+	"sync"
 
 	"golang.org/x/net/context"
 	"google.golang.org/grpc"
-	healthpb "google.golang.org/grpc/health/grpc_health"
+	"google.golang.org/grpc/codes"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1alpha"
 )
 
-// HealthCheck is the client side function to health-check a server
-func HealthCheck(t time.Duration, cc *grpc.ClientConn) error {
-	ctx, _ := context.WithTimeout(context.Background(), t)
-	hc := healthpb.NewHealthCheckClient(cc)
-	req := new(healthpb.HealthCheckRequest)
-	_, err := hc.Check(ctx, req)
-	return err
+type HealthServer struct {
+	// StatusMap stores the serving status of a service
+	statusMap map[string]int32
 }
 
-type HealthServer struct {
+func NewHealthServer() *HealthServer {
+	return &HealthServer{
+		statusMap: make(map[string]int32),
+	}
+}
+
+func (s *HealthServer) Check(ctx context.Context, in *healthpb.HealthCheckRequest) (out *healthpb.HealthCheckResponse, err error) {
+	service := in.Host + ":" + in.Service
+	out = new(healthpb.HealthCheckResponse)
+	status, ok := s.statusMap[service]
```

Comment:
This should be like:

if status, ok := s.statusMap[service]; ok {
  return &healthpb.HealthCheckResponse{...}, nil
}
return nil, grpc.Errorf(codes.NotFound, ....)